jbpm-human-task/jbpm-human-task-workitems/src/main/java/org/jbpm/services/task/wih/AbstractHTWorkItemHandler.java

```diff
@@ -224,12 +224,17 @@ protected ContentData createTaskContentBasedOnWorkItemParams(KieSession session,
         return data;
     }
     
-    protected boolean isAutoClaim(WorkItem workItem, Task task) {
-        String swimlaneUser = (String) workItem.getParameter("SwimlaneActorId");
-        if (swimlaneUser != null  && !"".equals(swimlaneUser) && task.getTaskData().getStatus() == Status.Ready) {
-            return true;
-        }
+    protected boolean isAutoClaim(KieSession session, WorkItem workItem, Task task) {
+        String autoclaim = (String) session.getEnvironment().get("Autoclaim");
```

Comment:
I think since this is user entered in some config relying on case-sensitivie and un-trimmed "false" can cause issues (user error ..entered extra space etc etc).
Maybe better to use Boolean.parseBoolean since its case insensitive and also use for example trim() or some StringUtil helper to remove spaces from string before checking.

Also I think "Autoclaim"  and "SwimlaneActorId " could  be constants somewhere instead of hardcoded. wdyt?

Reply:
@tsurdilo  @mswiderski  I did some changes in order to make case-insensitive the value of the variable and added a trim() too.